### DISCUSSION TITLE
EmrAddStepsOperator does not have an input for Execution Role ARN

### DISCUSSION THREAD

**Discussion by rankol** [apache/airflow#25683] [Ideas]
I've attempted to create this as a feature request but when I submit the Issue I get re-directed to a webpage that says `This site can't be reached` so I'm just posting this here for now.

The EmrAddStepsOperator does not have an input for Execution role ARNs.
https://github.com/apache/airflow/blob/bf14d14e272d24ee7a3f798c242593078359383e/airflow/providers/amazon/aws/operators/emr.py#L34
 
The Boto3 EMR client does have an option for this input.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr.html#EMR.Client.add_job_flow_steps
 
EMR version 6.7 recently released which allows for Execution role ARNs to be used with EMR job steps. This seems to be done at the CLI with `--execution-role-arn = $RUNTIME_ROLE_ARN`
https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-steps-runtime-roles.html

This would allow for EMR jobs to use an execution role for IAM permissions instead of using the EC2 instance profile attached to the cluster for permissions. Other EMR operators like the container operator or serverless operator allow for execution role ARNs to be passed in.

**Comment by Taragolis** [comment]
@rankol I would suggest two ways

**Option 1 Quick**:
Create own operator based on existed Operator in amazon-provider and make some changed, you just need add additional argument to `__init__` and add change one line in `execute` method of operator.

https://github.com/apache/airflow/blob/648e224cd455f1e374c58cfa48eb1c0ed69c698d/airflow/providers/amazon/aws/operators/emr.py#L111-L113

**Option 2**:
Contribute to apache-airflow-providers-amazon. Basically need to make the same changes and create some tests.
